### DISCUSSION TITLE
single quotes allowed for single characters

### DISCUSSION THREAD

**Discussion by beauxq** [astral-sh/ruff#13057] [Ideas]
To be friendly to people with habits from C/Java/Rust
I would like to enforce double quotes for all strings with 0 or 2 or more characters.
For strings with 1 character, either single quotes or double quotes is ok.

Is there any way to configure ruff this way?

**Comment by MichaReiser** [comment]
Hi @beauxq 

Ruff does support a few options that allow customizing the preferred quote styles, but you're looking for something more flexible. 
I understand the motivation: I could see myself typing `'a'`, but I would be okay if the formatter automatically changed the quotes to `". ` It also signals me that Python doesn't differentiate between strings and chars.
